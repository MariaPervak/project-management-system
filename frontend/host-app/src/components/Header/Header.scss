.header{
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-bottom: 1px solid var(--gray);
  padding: 25px;

  .logo{
    img{
      width: 40px;
      height: 40px;
    }
  }
}