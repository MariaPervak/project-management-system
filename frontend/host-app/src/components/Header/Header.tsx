import LogoImage from './logo.png';
import './Header.scss';

const Header = () => {
  return (
    <header className="header">
      <div className="logo"><img src={LogoImage} alt=""/></div>
      <input type="search" placeholder="Номер задачи"/>
      <div>profile</div>
    </header>
  );
};

export default Header;