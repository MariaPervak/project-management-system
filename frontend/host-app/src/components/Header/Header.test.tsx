import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from './Header';

jest.mock('../Auth/Auth', () => () => <div data-testid="auth-component">Auth</div>);
jest.mock('./logo_1.png', () => 'mock-logo-path.png');

describe('Header component', () => {
    beforeEach(() => {
        render(<Header />);
    });

    it('renders without crashing', () => {
        expect(screen.getByRole('banner')).toBeInTheDocument();
    });

    it('has correct header class', () => {
        expect(screen.getByRole('banner')).toHaveClass('header');
    });

    it('renders logo with correct text and image', () => {
        const logoContainer = screen.getByText('Система управления проектом').closest('.logo');
        expect(logoContainer).toHaveClass('logo');
        const logoDiv = screen.getByRole('banner').querySelector('.logo');
        expect(logoDiv).toHaveClass('logo');
        expect(logoDiv).toContainElement(screen.getByRole('presentation'));

        const imgElement = screen.getByRole('presentation');
        expect(imgElement).toHaveAttribute('src', 'mock-logo-path.png');
        expect(imgElement).toHaveAttribute('alt', '');
    });

    it('renders Auth component', () => {
        expect(screen.getByTestId('auth-component')).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Header />);
        expect(asFragment()).toMatchSnapshot();
    });
});