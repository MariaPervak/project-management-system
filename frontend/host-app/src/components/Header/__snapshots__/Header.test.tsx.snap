// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="logo"
    >
      <img
        alt=""
        src="mock-logo-path.png"
      />
       Система управления проектом
    </div>
    <div
      data-testid="auth-component"
    >
      Auth
    </div>
  </header>
</DocumentFragment>
`;
